src/grids/MasonryGrid.ts

```diff
@@ -166,13 +170,20 @@ export class MasonryGrid extends Grid<MasonryGridOptions> {
       gap,
       align,
     } = this.options;
+    const containerInlineSize = this.getContainerInlineSize();
     const columnSizeOption = this.columnSize || this.outlineSize;
-    const column = this.column || this.outlineLength || 1;
+    const columnOption = this.column || this.outlineLength;
+    let column = columnOption || 1;
 
     let columnSize = 0;
 
     if (align === "stretch") {
-      columnSize = (this.getContainerInlineSize() + gap) / (column || 1) - gap;
+      if (!columnOption) {
+        const maxStretchColumnSize = this.maxStretchColumnSize || Infinity;
```

Comment:
I got the point for the above case, but how about this case?
`Infinity` will be used instead if the user set `maxStretchColumnSize` to 0, I wonder if that case is possible though.

Reply:
@WoodNeck 

Yes. When the value is 0, it is set to Infinity.

Reply:
That looks quite weird... but okay then 👍